src/main/java/github/scarsz/discordsrv/listeners/DiscordConsoleListener.java

```diff
@@ -90,19 +94,29 @@ public void onGuildMessageReceived(GuildMessageReceivedEvent event) {
         if (logFile != null) {
             try {
                 FileUtils.writeStringToFile(
-                    logFile,
-                    "[" + TimeUtil.timeStamp() + " | ID " + event.getAuthor().getId() + "] " + event.getAuthor().getName() + ": " + event.getMessage().getContentRaw() + System.lineSeparator(),
-                    StandardCharsets.UTF_8,
-                    true
+                        logFile,
+                        "[" + TimeUtil.timeStamp() + " | ID " + event.getAuthor().getId() + "] " + event.getAuthor().getName() + ": " + event.getMessage().getContentRaw() + System.lineSeparator(),
+                        StandardCharsets.UTF_8,
+                        true
                 );
             } catch (IOException e) {
                 DiscordSRV.error(LangUtil.InternalMessage.ERROR_LOGGING_CONSOLE_ACTION + " " + logFile.getAbsolutePath() + ": " + e.getMessage());
                 if (DiscordSRV.config().getBoolean("CancelConsoleCommandIfLoggingFailed")) return;
             }
         }
 
+        // Create DiscordConsoleCommandPreProcessEvent
+        DiscordConsoleCommandPreProcessEvent consoleEvent = DiscordSRV.api.callEvent(new DiscordConsoleCommandPreProcessEvent(event, event.getMessage().getContentRaw(), true));
+
+        //stop the command from being run if the API user wants to stop the command from being run
+        if (consoleEvent.isCancelled()) return;
+
+        // It now uses the command that comes out of the event, in case the api user changes it
```

Comment:
```suggestion
        DiscordConsoleCommandPreProcessEvent consoleEvent = DiscordSRV.api.callEvent(new DiscordConsoleCommandPreProcessEvent(event, event.getMessage().getContentRaw(), true));

        // Stop the command from being run if a API user cancels the event
        if (consoleEvent.isCancelled()) return;

        // It now uses the command that comes out of the event, in case a api user changes it
```